jest-common/src/main/java/io/searchbox/core/Cat.java

```diff
@@ -0,0 +1,92 @@
+package io.searchbox.core;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import io.searchbox.action.AbstractAction;
+import io.searchbox.action.AbstractMultiIndexActionBuilder;
+import io.searchbox.action.AbstractMultiTypeActionBuilder;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+
+/**
+ * @author Bartosz Polnik
+ */
+public class Cat extends AbstractAction<CatResult> {
+    private final static String PATH_TO_RESULT = "result";
+    public Cat(AbstractAction.Builder<Cat, ? extends Builder> builder) {
+        super(builder);
+    }
+
+    Cat() {
+    }
+
+    @Override
+    public String getRestMethodName() {
+        return "GET";
+    }
+
+    @Override
+    public String getPathToResult() {
+        return PATH_TO_RESULT;
+    }
+
+    @Override
+    public CatResult createNewElasticSearchResult(String responseBody, int statusCode, String reasonPhrase, Gson gson) {
+        return createNewElasticSearchResult(new CatResult(gson), responseBody, statusCode, reasonPhrase, gson);
+    }
+
+    @Override
+    protected JsonObject parseResponseBody(String responseBody) {
+        JsonObject result = new JsonObject();
+        if (responseBody != null && !responseBody.trim().isEmpty()) {
+            result.add(PATH_TO_RESULT, new JsonParser().parse(responseBody).getAsJsonArray());
+        }
+
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (obj == this) {
+            return true;
+        }
+        if (obj.getClass() != getClass()) {
+            return false;
+        }
+
+        return new EqualsBuilder()
+                .appendSuper(super.equals(obj))
+                .isEquals();
+    }
+
+    public static class IndicesBuilder extends AbstractMultiTypeActionBuilder<Cat, IndicesBuilder> {
+        public IndicesBuilder() {
+            setHeader("content-type", "application/json");
+        }
+
+        @Override
+        public Cat build() {
+            Cat cat = new Cat(this);
+            String uriSuffix = cat.buildURI();
```

Comment:
URI build logic does not fit in the builder itself. Instead I'd do *roughly* sth like this:

```java
public class Cat extends AbstractAction<CatResult> {
    private final static String PATH_TO_RESULT = "result";
    private final String operationPath;

    public Cat(AbstractAction.Builder<Cat, ? extends Builder> builder) {
        super(builder);
        this.operationPath = builder.operationPath;
        setURI(buildURI());
    }

    @Override
    protected String buildURI() {
        String uriSuffix = super.buildURI();
        return "_cat/aliases" + (uriSuffix.isEmpty() ? "" : "/") + uriSuffix;
    }

// .... adjust builder classes and types (and obviously Cat constructor) to provide operationPath string
```